client/profile-wizard/steps/business-details/flows/selective-bundle/selective-extensions-bundle/index.js

```diff
@@ -327,29 +373,64 @@ export const SelectiveExtensionsBundle = ( {
 	productTypes,
 } ) => {
 	const [ showExtensions, setShowExtensions ] = useState( false );
-	const [ values, setValues ] = useState( {} );
+	const [ values, setValues ] = useState( baseValues );
+	const [ installableExtensions, setInstallableExtensions ] = useState(
+		initialInstallableExtensions
+	);
+	const [ isFetching, setIsFetching ] = useState( true );
+
+	const allowMarketplaceSuggestions = useSelect( ( select ) =>
+		select( SETTINGS_STORE_NAME ).getSetting(
+			'wc_admin',
+			'allowMarketplaceSuggestions'
+		)
+	);
 
 	useEffect( () => {
-		const initialValues = installableExtensions.reduce(
-			( acc, curr ) => {
-				const plugins = getVisiblePlugins(
-					curr.plugins,
-					country,
-					industry,
-					productTypes
-				).reduce( ( pluginAcc, { slug } ) => {
-					return { ...pluginAcc, [ slug ]: true };
-				}, {} );
+		const setLocalInstallableExtensions = () => {
+			const initialValues = createInitialValues(
+				installableExtensionsData,
+				country,
+				industry,
+				productTypes
+			);
+			setInstallableExtensions( installableExtensionsData );
+			setValues( initialValues );
+			setIsFetching( false );
+		};
 
-				return {
-					...acc,
-					...plugins,
-				};
-			},
-			{ install_extensions: true }
-		);
-		setValues( initialValues );
-	}, [ country ] );
+		if (
+			window.wcAdminFeatures &&
+			window.wcAdminFeatures[ 'remote-extensions-list' ] === true &&
+			allowMarketplaceSuggestions
+		) {
+			apiFetch( {
+				path: '/wc-admin/onboarding/free-extensions',
+			} )
```

Comment:
I don't mind the use of `apiFetch` here, but is there a reason of using this instead of a redux store? You could move this logic to a resolver (potentially).

If sticking with `apiFetch` we should add a check to see if the fetch has already happened, so it doesn't refetch every-time we switch tabs back to free features.

Reply:
>we should add a check to see if the fetch has already happened

If `SelectiveExtensionsBundle` gets remounted every time the "Free extensions" tab is clicked, how could we keep track of the data? We'd need to cache it somehow.... like a store 😆 . Any ideas here?

> is there a reason of using this instead of a redux store? 

No particular reason other than it being an endpoint that is barely used. Its only one small part of the OBW that many won't even click. Do you think its worthwhile for data persistence to make sure its only called once. For me, I don't think users will switch back and forth all the time between tabs, but I'm open to adding a store if there is an argument I'm missing.

Reply:
> If SelectiveExtensionsBundle gets remounted every time the "Free extensions" tab is clicked, how could we keep track of the data? We'd need to cache it somehow.... like a store 😆 . Any ideas here?

True, I guess a store would be the best option (if we wanted to prevent re-fetching), which doesn't quite seem necessary with your answer after that.

> Do you think its worthwhile for data persistence to make sure its only called once? For me, I don't think users will switch back and forth all the time between tabs, but I'm open to adding a store if there is an argument I'm missing.

I agree, it probably won't be necessary. I suppose the only case would be if we wanted to pre-fetch it to prevent a loading screen, I didn't notice any loading lack when testing, but that might be because of the helper plugin?
We can stick with your current implementation I think.